### PR TITLE
Added initial data validation

### PR DESCRIPTION
**Pull request by bartr**
## Type of PR

- [X] Documentation changes
- [X] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [X] GitHub Template changes

## Purpose of PR
- synched github templates with helium
- added param validation and examples to readme
- added initial param validation
- addressed design review feedback

## Issues Closed or Referenced
- References #14

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,227 @@ docker run -it --rm retaildevcrew/webvalidate --host https://www.microsoft.com -
...
+      - valid: cannot be combined with MinCout or MaxCount
+    - MinCount
+      - minimum number of items
+      - valid: >= 0
+        - can be comined with MaxCount
+    - MaxCount
+      - maximum number of items
+      - valid: > MinCount
+        - can be comined with MinCount
+    - ForEach[JsonObject]
```
Comment (aflinchb) [inline README.md]: Typo here and above "comined" not "combined"

File: src/WebValidation/Model/Request.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            // null check
+            if (Validation.JsonArray == null)
+            {
+                message = "jsonArray: cannot validate null request";
+                return false;
+            }
+
+            // must be >= 0 (0 is default
+            if (Validation.JsonArray.Count < 0 || Validation.JsonArray.MinCount < 0 || Validation.JsonArray.MaxCount < 0)
+            {
```
Comment (aflinchb) [inline src/WebValidation/Model/Request.cs]: Is the default set to 0 because of the default serialization behavior?

File: src/TestFiles/bad.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  {"Url":"/api/movies?pagesize=5&pagenumber=2","PerfTarget":{"Category": "SearchMovies"},"Validation":{"Code":200,"ContentType":"application/json","Contains": [{"Value": "tt0115438"}],"JsonArray":{"Count":5}}},
+
+  {"Url":"/api/movies/ab0120737","PerfTarget":{"Category": "DirectRead"},"Validation":{"Code":400,"ContentType":"text/plain"}},
+  {"Url":"/api/movies/tt123","PerfTarget":{"Category": "DirectRead"},"Validation":{"Code":400,"ContentType":"text/plain"}},
+  {"Url":"/api/movies/foo","PerfTarget":{"Category": "DirectRead"},"Validation":{"Code":400,"ContentType":"text/plain"}},
+
+  {"Url":"/api/movies?actorid=nm123","PerfTarget":{"Category": "SearchMovies"},"Validation":{"Code":400,"ContentType":"text/plain"}},
+  {"Url":"/api/movies?actorid=ab123","PerfTarget":{"Category": "SearchMovies"},"Validation":{"Code":400,"ContentType":"text/plain"}},
+  {"Url":"/api/movies?actorid=foo","PerfTarget":{"Category": "SearchMovies"},"Validation":{"Code":400,"ContentType":"text/plain"}},
+]
```
Comment (jyotsnaravikumar) [inline src/TestFiles/bad.json]: we can add the following cases here 
/api/movies?toprated=foo
/api/movies?rating=foo
/api/movies?rating<=0
/api/movies?rating>=10
/api/movies?rating<=-1

/api/actors?pageNumber=-1
/api/actors?pageNumber=0
/api/actors?pageSize=-1
/api/actors?pageSize=0

File: src/WebValidation/Model/Request.cs (hunk 2 above)
Comment (bartr) [reply src/WebValidation/Model/Request.cs]: Currently yes. Plan is to convert to int? in a future PR, so this isn't a complete validation

File: src/TestFiles/bad.json (hunk 3 above)
Comment (bartr) [reply src/TestFiles/bad.json]: future PR

File: README.md [hunk 4]
```diff
@@ -11,6 +11,227 @@ docker run -it --rm retaildevcrew/webvalidate --host https://www.microsoft.com -
...
+    [
+      { "Value": "User-agent: *" },
+      { "Value": "Disallow: /en-us/windows/si/matrix.html"}
+    ]
+  }
+}
+```
+
+## Sample helium tests
+
```
[the remark this answers is not in the document]
Comment (bartr) [reply README.md]: added

**Pull request merged**